torch/multiprocessing/reductions.py

```diff
@@ -118,6 +118,20 @@ def rebuild_tensor(cls, storage, metadata):
     return t
 
 
+def rebuild_tensor_nested(cls, storage, metadata):
```

Comment:
Can we re-use the one from serialization.py?

Reply:
For CUDA NTs, I think we need to use the CUDA sharing logic for all this to work properly. I updated the PR with proper CUDA tests and changed the logic here to recursively call into `reduce_tensor()` for the NT components -> rebuild appropriately. Please lmk if you think there is some reuse opportunity I'm missing after this.